packages/plugins/src/cli/install/index.ts

```diff
@@ -14,12 +14,35 @@ export const handleInstall = async () => {
 
   const dependencyInstall = await spawnAsync('forge', args)
   if (dependencyInstall.code !== 0) {
-    // The `stdout` contains the trace of the error.
-    console.log(dependencyInstall.stdout)
-    // The `stderr` contains the error message.
-    console.log(dependencyInstall.stderr)
-    process.exit(1)
+    // If installation fails due to missing tage, then run forge update and attempt again
+    if (
+      dependencyInstall.stderr.includes(
```

Comment:
I think relying on an error message thrown by Foundry to determine when we should run `forge update` is brittle. If they ever change the error message, this code will break. Is there another way to check whether the tag is missing locally? Or, maybe we could just always run `forge update`?

Reply:
I generally agree and spent time looking into this. I feel very strongly that we should just check the error message in this case for the following reasons:
- `forge update` errors if you do not have the dependency installed
- forge does not provide a reliable way to determine what dependencies are installed
- checking for the existence and/or contents of the `.gitmodules` file is not a reliable way to determine if the project installed (because there may be multiple .gitmodules files which are for different nested repos depending on the users own use of git submodules)
- checking for the existence of the library in the users project is not a reliable way to determine if it's installed

So the only way to determine if the user already has the library installed would be to search the git commit history. All around, I think pretty much anything we implement to try and determine if the library is already installed will be far more error-prone than just using the error message.

Reply:
I don't think there is another practical solution for this, mainly because Forge does not provide a reliable way to detect what dependencies are installed. If we wanted to detect this another way we'd have to do one of the following:
- Search for the `sphinx` folder in the users `lib` folder. This is not reliable because the user may remove the `lib/sphinx` folder, or it maybe removed automatically by forge (due to a failed dependency installation for example)
- Search for the `.gitmodules` file and check if it contains the sphinx library. This is not reliable because there may be multiple `.gitmodules` files in the project due to the use of submodels outside of just how foundry uses them.
- Search through the git commit history for evidence of the library being installed. I did not try to implement this, but it seems like it would be very error prone as well.

So I don't think there's an alternative to checking the error message that is not even more fragile. I think the only real alternative would be to simply remove this command and instead ask the user to install the library themselves, but I don't think we should do that either because humans are also not that reliable.

Reply:
Why not always run `forge install` then `forge update sphinx-labs/sphinx`?

Reply:
To be clear, I'm asking why we can't always run these commands instead of just running them if an error occurs during the initial `forge install` command

Reply:
`forge update sphinx-labs/sphinx` does not work if you do not have the library installed. We could use it, but then we would need to catch the error and match on the message for that instead.

Reply:
Why would we need to catch the error message? I.e. can't we do this?
1. Run `forge install`
2. Run `forge update sphinx-labs/sphinx`, wrapping it in a try/catch. Don't do anything if it fails.
3. Run `forge install sphinx-labs/sphinx@${CONTRACTS_LIBRARY_VERSION} --no-commit`

I understand that you've tested this extensively, and I'm sure this approach works now. My concern is that i'll break in the future.

Reply:
`forge update `sphinx-labs/sphinx` could fail for many reasons. It is unsafe to catch and do nothing.

Reply:
What about using `git submodule status` to check if the sphinx dependency is already installed?

Reply:
I'm ok with using that if you're more comfortable with it. I've updated this PR to work that way.

Reply:
My general opinion on `git submodule status` is that it's probably also subject to bugs related to the user modifying their `.gitmodules` file manually and/or having multiple `.gitmodules` files in the same repo. However, I don't think it's any less likely to produce bugs than forge updating their error message, so I'm cool with it.